scripts/assert_no_yarn_warnings.sh

```diff
@@ -10,16 +10,16 @@ MATCH=("warning")
 # There are few yarn warnings we can't find a fix for. Excluding those. 
 # TODO: Keep checking occasionally if the warnings are fixed upstream.
 EXCLUDE=("Pattern \[\".*\"\] is trying to unpack in the same destination")
```

Comment:
Could not find a solution to fix this warning, so why have to disable it from lint script.

Reply:
Unmet peer dependency warnings seem to be resolved but there are further warnings. I quickly looked at how to resolve those warnings before, I linked to some issues [here](https://discord.com/channels/593655374469660673/605818244036821012/1169604465339420733) but does not seem like there are quick solutions for it.

```
[4/5] Linking dependencies...
warning Workspaces can only be enabled in private projects.
warning Workspaces can only be enabled in private projects.
```

Lint run did not properly catch the warnings, see [job run](https://github.com/ChainSafe/lodestar/actions/runs/7142330071/job/19451491587?pr=6172#step:5:9).

Reply:
Related to workspace warning, I am seeing it on Mac for long time. It's a known issue and non-fixable in yarn 1. 

https://github.com/yarnpkg/yarn/issues/8580#issuecomment-1734464145

Reply:
After further investigation, it turns out both warnings are related to https://github.com/davidmyersdev/vite-plugin-node-polyfills, which we use here
https://github.com/ChainSafe/lodestar/blob/8bd19f431f4766bbb8066a6d858cb756d68536ae/vitest.base.browser.config.ts#L4

There `warning Pattern ["buffer@^6.0.3"] is trying to unpack in the same destination "/home/nico/.cache/yarn/v6/npm-buffer-polyfill-6.0.3-2ace578459cc8fbe2a70aaa8f52ee63b6a74c6c6-integrity/node_modules/buffer-polyfill" as pattern ["buffer-polyfill@npm:buffer@^6.0.3"]. This could result in non-deterministic behavior, skipping.` happens because of
https://github.com/davidmyersdev/vite-plugin-node-polyfills/blob/bfb529025d722a49cf98c395550fd46454d0de39/package.json#L73

And `warning Workspaces can only be enabled in private projects.` happens due to 
https://github.com/davidmyersdev/vite-plugin-node-polyfills/blob/bfb529025d722a49cf98c395550fd46454d0de39/package.json#L94-L96

I haven't looked further on why we need those polyfills but shouldn't our code work in the browser without them? What if we run the code without polyfills in production will it break?

Reply:
Have tried to remove `vite-plugin-node-polyfills` package but so far unsuccessful to make the browser tests work (https://github.com/ChainSafe/lodestar/compare/unstable...nflaig/fix-yarn-warnings). Tried few different things but none worked, what I didn't try yet is using https://github.com/niksy/node-stdlib-browser#vite directly, this way we can avoid the polyfill package.

Opened two issues on polyfills package, maybe they can resolve it on their end
- https://github.com/davidmyersdev/vite-plugin-node-polyfills/issues/60
- https://github.com/davidmyersdev/vite-plugin-node-polyfills/issues/61

Reply:
We need to use the polyfills for the browsers as Nodejs modules are not available in browser environment. 

We can try few other polyfill packages, but using browsified modules directly meant to have multiple dependencies a a lot of vite configuration to manage in our code.